core/src/Lender.sol

```diff
@@ -119,9 +119,13 @@ contract Lender is Ledger {
         shares = _convertToShares(amount, inventory, cache.totalSupply, /* roundUp: */ false);
         require(shares != 0, "Aloe: zero impact");
 
-        // Ensure tokens were transferred
+        // Ensure tokens are transferred
+        ERC20 asset_ = asset();
         cache.lastBalance += amount;
-        require(cache.lastBalance <= asset().balanceOf(address(this)), "Aloe: insufficient pre-pay");
+        bool didPrepay = cache.lastBalance <= asset_.balanceOf(address(this));
```

Comment:
I understand now - if the user pre-pays using the `Router`, then that means that [this check](https://github.com/aloelabs/aloe-ii/pull/87/files#diff-bd82376927d64137b16c2eb48179808b927a9644b06517dfb511e02bdce676ecR125) will return `true`, thus only pre-pay is supported through this flow. ✅